src/commands/synthetics/run-tests-lib.ts

```diff
@@ -45,6 +46,7 @@ type ExecuteOptions = {
 
 export const executeTests = async (
   reporter: MainReporter,
+  appUploadReporter: AppUploadReporter,
```

Comment:
I'd like to avoid changing this function's signature at all costs, because it would be a breaking change for the folks who use `@datadog/datadog-ci` programmatically, as a library.

And since this app upload reporter is short-lived and only used for Mobile tests, I think we should construct it directly in `uploadMobileApplicationsAndOverrideConfigs()` instead of passing it around. WDYT?